test/jdk/java/nio/file/attribute/BasicFileAttributeView/CreationTime.java

```diff
@@ -118,6 +129,18 @@ static void test(Path top) throws IOException {
         }
     }
 
+    // Reflective implementation so as to compile on non-Linux platforms
+    private static boolean unixDispatcherSupportsBirthTime() {
+        try {
+            Class<?> unixDispatcher = Class.forName(UNIX_DISPATCHER_CLASSNAME);
+            Method supportsBirthTime = unixDispatcher.getDeclaredMethod(SUPPORTS_BIRTH_TIME_METHOD);
+            supportsBirthTime.setAccessible(true);
+            return (boolean)supportsBirthTime.invoke(null);
+        } catch (Throwable e) {
+            throw new RuntimeException("Illegal state!");
```

Comment:
I agree with Brian, we need to find a better way to do this.

Reply:
Sure, happy to. Any suggestions?

Reply:
I'm experimenting with a native function in a test helper. Hopefully that'll be more amenable.